### PR TITLE
1ES Templates onboarding

### PR DESCRIPTION
**Pull request by YuliiaKovalova**
Split the pipeline on production and PR one.
Onboard 1es official templates for prod yml, validated internally https://dev.azure.com/dnceng/internal/_build/results?buildId=2404188&view=results

On azure devops side everything is ready for new naming.

### CODE REVIEW

File: azure-pipelines-PR.yml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+          displayName: Run Tests
+          inputs: 
+            command: test
+
+  - job: LinuxTests
+    timeoutInMinutes: 30
+    container: UbuntuContainer
+    pool:
+      name: NetCore-Public
+      demands: ImageOverride -equals Build.Ubuntu.1804.Amd64.Open
```
Comment (baronfel) [inline azure-pipelines-PR.yml:52]: this is a pretty old version of Ubuntu - at what point should we update?

Comment (rainersigwald) [reply azure-pipelines-PR.yml]: Follow-up question: _should_ this be running tests on like a min-spec OS?

File: azure-pipelines-official.yml [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+        - task: DotNetCoreCLI@2
+          displayName: Run Tests
+          inputs: 
+            command: test
+
+      - job: MacOSTests
+        timeoutInMinutes: 30
+        pool: 
+          name: Azure Pipelines
+          image: macOS-11
```
Comment (rainersigwald) [inline azure-pipelines-official.yml:120]: Similarly this is pretty old, _should_ it be?

**Review by rainersigwald — APPROVED** [review]

File: azure-pipelines-PR.yml (hunk 1 above)
Comment (YuliiaKovalova) [reply azure-pipelines-PR.yml]: looking at vstest pipelines state, the image selection seems to be reasonable: 
https://github.com/microsoft/vstest/blob/main/azure-pipelines.yml

File: azure-pipelines-official.yml (hunk 2 above)
Comment (YuliiaKovalova) [reply azure-pipelines-official.yml:120]: I have double checked with the 1es team - it's ok to bump it.

**Review by baronfel — APPROVED** [review]

**Pull request merged**